### PR TITLE
Re-structure Readme using a TOC

### PR DESCRIPTION
**Pull request by jeffwidman**
Over time the Readme has grown organically by adding new sections as needed. Unfortunately, the length and relative lack of order makes it not very approachable for folks new to the project.

So this adds a Table of Contents structured in a way that's more accessible, and then re-orders the content appropriately.

[rendered](https://github.com/dependabot/dependabot-core/blob/re-order-readme-using-intuitive-TOC/README.md)

### CODE REVIEW

File: README.md
Comment (jeffwidman) [inline README.md:25]: I think this is a reasonable TOC? 

I purposefully omitted some of the subheadings, in order to focus on the main entrypoints that folks will be looking for when they first hit the readme...

File: README.md
Comment (jeffwidman) [inline README.md:472]: Using `"Dependabot Core"` seemed ambiguous to me... we always refer to it as a single compound word, so I renamed it to `"Dependabot-Core"` everywhere...

File: README.md
Comment (jeffwidman) [inline README.md:479]: All the hyperlinks will need to be audited once the contents are solid.

File: README.md
Comment (jeffwidman) [inline README.md:81]: @deivid-rodriguez I could use your input here... it seems generally we are good if they provide a public repository link, but if there's anything more specific that we really need please let me know.

I'm planning to hyperlink directly to this section from the issue template, and we'll probably want to add automation that auto-comments with a link to this bullet point when a maintainer applies a label indicating an issue is missing a link to a public repo.

File: README.md
Comment (jeffwidman) [inline README.md:109]: This is duplicated here and in the Contributing section... Kinda like to drop one of them, but not sure which one... I do like having a contribution workflow front and center on the readme.

File: README.md
Comment (jeffwidman) [inline README.md:136]: We used to have this shortcut buried further down, but we want new contributors to get going as fast as possible, so highlighting how to pull the image as the first step will save them 5-10 mins and lead to a much more pleasant first experience. At least for me it did.

File: README.md
Comment (jeffwidman) [inline README.md:143]: Anyone got an ARM machine to test? I don't have an M1 mac anymore...

File: README.md
Comment (jeffwidman) [inline README.md:83]: This is a new section, but I really like the go team's wiki page of examples... and if we were to start being more transparent about what we're currently working on, then folks wouldn't need to ask "any update here?"... would cut down on the noise a little.

**Review by jeffwidman — COMMENTED** [review]
Annotating the changes.

I'm marking as "ready" because ready for the first round of reviews. I still need to hyperlink it internally before we actually merge it.

File: README.md
Comment (deivid-rodriguez) [reply README.md:25]: I like it!

File: README.md
Comment (deivid-rodriguez) [inline README.md:77]: or the CLI?

File: README.md
Comment (deivid-rodriguez) [reply README.md:81]: I think it's fine. Obviously for fixing purposes and for writing better specs, the simpler a public repro is, the better. But it's hard for users (even for us!) to figure out what can be reduced/simplified so that a bug still reproduces. So I think a public repro is a good compromise.

File: README.md
Comment (deivid-rodriguez) [reply README.md:83]: I love it!

File: README.md
Comment (deivid-rodriguez) [inline README.md:365]: In the containers, there's a single version installed for each gem. So I think we can simplify this and the above commands to remove `bundle exec`.

**Review by deivid-rodriguez — APPROVED** [review]
I really love this update, it makes the README so much useful!

**Comment by Nishnha** [comment]
I like the table of contents idea, but it turns out README file already have one? https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/

**Comment by jeffwidman** [comment]
> I like the table of contents idea, but it turns out README files already have one? https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/

That's awesome, I had _no_ idea that existed.

I still think we should have an explicit TOC of the high-level links because it's much more discoverable than the built-in one.

I wish they offered a TOC widget that we could embed for those who want an always-visible TOC.

File: README.md
Comment (jeffwidman) [reply README.md:81]: SG, we can always tweak what we ask for down the road...

File: README.md
Comment (jeffwidman) [reply README.md]: Added some Arm-specific notes based on insights from @pavera

**Comment by jeffwidman** [comment]
Okay, I've addressed all the TODO's, which were mostly missing hyperlinks.

Feedback both here and on our internal slack was very positive. I'm sure this can be improved further, but by landing it now we can start deep-linking to various sections when responding to issues.

So I'm going to go ahead and merge, and then we can add further refinements in follow-on PR's.

**Pull request merged**
